charts/opentelemetry-operator/README.md

```diff
@@ -0,0 +1,319 @@
+
+The OpenTelemetry Collector CRD created by this chart won't be removed by default and should be manually deleted:
+
+```console
+$ kubectl delete crd opentelemetrycollectors.opentelemetry.io
+```
+
+If the namespace is created by Helm instead of yourself, please skip this step. If you created the namespace and set
+`createNamespace` to `false`, you should remove it manually:
+
+```console
+$ kubectl delete ns opentelemetry-operator-system
+```
+
+## Upgrade Chart
+
+```console
+$ helm upgrade my-opentelemetry-operator open-telemetry/opentelemetry-operator
+```
+
+Please note that by default, the chart will be upgraded to the latest version. If you want to upgrade to a specific version,
+use `--version` flag.
+
+With Helm v3.0, CRDs created by this chart are not updated by default and should be manually updated.
+Consult also the [Helm Documentation on CRDs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions).
+
+_See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
+
+## Configuration
+
+The following command will show all the configurable options with detailed comments.
+
+```console
+$ helm show values open-telemetry/opentelemetry-operator
+```
+
+## Install OpenTelemetry Collector
+
+_See [OpenTelemetry website](https://opentelemetry.io/docs/collector/) for more details about the Collector_
+
+Once the opentelemetry-operator deployment is ready, you can deploy OpenTelemetry Collector in our Kubernetes
+cluster.
+
+The Collector can be deployed as one of four modes: Deployment, DaemonSet, StatefulSet and Sidecar. The default
+mode is Deployment. We will introduce the benefits and use cases of each mode as well as giving an example for each.
+
+### Deployment Mode
+
+If you want to get more control of the OpenTelemetry collector and create a standalone application, Deployment would
+be your choice. With Deployment, you can relatively easily scale up the collector to monitor more targets, roll back
+to an early version if anything unexpected happens, pause the collector, etc. In general, you can manage your collector
+instance just as an application.
+
+The following example configuration deploys the Collector as Deployment resource. The receiver is jaeger-receiver and
+the exporter is logging exporter.
+
+```console
+$ kubectl apply -f - <<EOF
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: my-collector
+spec:
+  mode: deployment # This configuration is omittable.
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+    processors:
+
+    exporters:
+      logging:
+
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: []
+          exporters: [logging]
+EOF
+```
+
+### DaemonSet Mode
+
+DaemonSet should satisfy your most basic needs and is the most common method to deploy OpenTelemetry collector.
```

Comment:
```suggestion
DaemonSet should satisfy your most basic needs and is the most common method to deploy OpenTelemetry Collector.
```

Is it, though? I would love to hear more about it :-)

Reply:
Here is my thought - please correct me if I'm wrong. Installing Collector as DaemonSet to the k8s cluster basically means you are installing an agent and every k8s node has its own Collector copy. And I guess the most common needs would be using the Collector as an agent to scrape the telemetry data. Just my 2 cents.

Reply:
My contention is that you are claiming that daemonsets are the most common method of deploying the collector. I don't think I've seen data about it, but if you can back it, I'm all for keeping this. Note that the OpenTelemetry isn't only about scraping data, but also about receiving data, especially when we talk about traces. In that respect, sidecars are common and popular, as they are a buffer between the workload and the remote collector, and an outage in the collector at the node level won't affect all pods on that node.

Reply:
"the most common method" is absolute. I will rephrase this statement to clarify that in what situation, DaemonSet would be a good choice.

Yes, as you said, Collector has multiple receivers to ingest data. We should be clear to users what mode is best for what.